### PR TITLE
chore: add asserNever

### PR DESCRIPTION
**Pull request by nytamin**
This PR adds a few assertNever in various places.

### CODE REVIEW

File: meteor/client/ui/Settings/components/ConfigManifestEntryComponent.tsx [hunk 1]
```diff
@@ -61,6 +62,14 @@ export const renderEditAttribute = (
 				}
 			/>
 		)
+	// } else if (configField.type === ConfigManifestEntryType.TABLE) {
```
Comment (nytamin) [inline meteor/client/ui/Settings/components/ConfigManifestEntryComponent.tsx:65]: I've commented out these changes, as I wasn't sure on the best ways to address these

File: meteor/lib/collections/PeripheralDevices.ts [hunk 2]
```diff
@@ -145,6 +145,15 @@ export function getExternalNRCSName(device: PeripheralDevice | undefined): strin
...
 				return 'Google Sheet'
+			} else if (
+				device.type === PeripheralDeviceAPI.DeviceType.PLAYOUT  ||
+				device.type === PeripheralDeviceAPI.DeviceType.MEDIA_MANAGER ||
+				device.type === PeripheralDeviceAPI.DeviceType.PACKAGE_MANAGER
+			) {
+				// These aren't ingest gateways
+				return 'N/A'
+			} else {
+				assertNever(device.type)
```
Comment (jstarpl) [inline meteor/lib/collections/PeripheralDevices.ts:156]: Considering that it's possible to connect to Sofie using a custom peripheralDevice, even if the typeguard is useful, it should still try and return some useful value, perhaps `Unknown: ${device.type}`?

File: meteor/client/lib/SettingsNavigation.tsx [hunk 3]
```diff
@@ -18,6 +19,8 @@ export class SettingsNavigation extends React.Component<ISettingsNavigation> {
 			return <ShowStyle {...this.props} />
 		} else if (this.props.type === 'newshowstyle') {
 			return <NewShowStyle {...this.props} />
+		} else{
```
Comment (jstarpl) [inline meteor/client/lib/SettingsNavigation.tsx:22]: Linting error?

**Review by jstarpl — APPROVED** [review]
I really like what you did, one comment would be to think where it's actually physically possible that the type guard is hit in runtime (like when the data is provided by an external system) and either throw an error or provide a sensible fallback.

File: meteor/lib/collections/PeripheralDevices.ts (hunk 2 above)
Comment (nytamin) [reply meteor/lib/collections/PeripheralDevices.ts:156]: Good insight! Will fix!

**Pull request merged**